<a href="https://colab.research.google.com/github/xkaycc/SALES-PREDICTION/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Project 2**

### **Task**
- propose two possible datasets
- choose your first choice data set, and a backup data set 

**First and second choice: dataset 1&2**

1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person?  A business?  An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

## **Dataset 1**

**This dataset is titled medical cost insurance data and I will be using it for regression analysis** 

In [ ]:
##Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [ ]:
#load dataset 1 (insurance data)
df_1 = pd.read_csv(r"C:\Users\USER\Desktop\insurance.csv")

In [ ]:
#check first 5 rows of df_1
df_1.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [ ]:
#check the shape of df_1
df_1.shape
print (f' There are {df_1.shape[0]} rows and {df_1.shape[1]} columns')

 There are 1338 rows and 7 columns


## **Source of the Dataset**

**This dataset was gotten from kaggle site**
https://www.kaggle.com/datasets/mirichoi0218/insurance

## **Brief description of data**

**This dataset is having the data of 1K+ medical insurance and can we accurately predict insurance costs?**

- Dataset Walkthrough
- Understanding Dataset Hierarchy
- Data Preprocessing
- Exploratory Data Analysis
- Data Visualization
- Making Recommendation System

## **What is the target?**

- charges

## **What does one row represent?**

- Each row represent persons who has come to the hospital

## **Is this a classification or regression problem?**

- Regression

## **How many features does the data have?**

- There are six(6) features aside the target variable

## **How many rows are in the dataset?**

- 1338 rows

## **What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?**

In [ ]:
## Check for missing values
df_1.isnull().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

- no missing data

In [ ]:
## Have a look at the descriptive statistics
df_1.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

## **Summary**

- From the little analysis, I don't think i should encounter a major challenge during the course of the work. 
- No much cleaning and exploration is needed

# **Dataset 2**

**This dataset is titled hotel reservations data and I will be using it for classification analysis**

In [ ]:
#load dataset 1 (insurance data)
df_2 = pd.read_csv(r"C:\Users\USER\Desktop\Hotel Reservations.csv")

In [ ]:
#check first 5 rows of df_1
df_2.head()

Booking_ID  no_of_adults  no_of_children  no_of_weekend_nights  \
0   INN00001             2               0                     1   
1   INN00002             2               0                     2   
2   INN00003             1               0                     2   
3   INN00004             2               0                     0   
4   INN00005             2               0                     1   

   no_of_week_nights type_of_meal_plan  required_car_parking_space  \
0                  2       Meal Plan 1                           0   
1                  3      Not Selected                           0   
2                  1       Meal Plan 1                           0   
3                  2       Meal Plan 1                           0   
4                  1      Not Selected                           0   

  room_type_reserved  lead_time  arrival_year  arrival_month  arrival_date  \
0        Room_Type 1        224          2017             10             2   
1        Room_Type 1          5          2018             11             6   
2        Room_Type 1          1          2018              2            28   
3        Room_Type 1        211          2018              5            20   
4        Room_Type 1         48          2018              4            11   

  market_segment_type  repeated_guest  no_of_previous_cancellations  \
0             Offline               0                             0   
1              Online               0                             0   
2              Online               0                             0   
3              Online               0                             0   
4              Online               0                             0   

   no_of_previous_bookings_not_canceled  avg_price_per_room  \
0                                     0               65.00   
1                                     0              106.68   
2                                     0               60.00   
3                                     0              100.00   
4                                     0               94.50   

   no_of_special_requests booking_status  
0                       0   Not_Canceled  
1                       1   Not_Canceled  
2                       0       Canceled  
3                       0       Canceled  
4                       0       Canceled

In [ ]:
#check the shape of df_1
df_2.shape
print (f' There are {df_2.shape[0]} rows and {df_2.shape[1]} columns')

 There are 36275 rows and 19 columns


## **Source of the Dataset**

This dataset was gotten from kaggle site https://www.kaggle.com/datasets/ahsan81/hotel-reservations-classification-dataset

## **Brief description of data**

The online hotel reservation channels have dramatically changed booking possibilities and customers’ behavior. A significant number of hotel reservations are called-off due to cancellations or no-shows. The typical reasons for cancellations include change of plans, scheduling conflicts, etc. This is often made easier by the option to do so free of charge or preferably at a low cost which is beneficial to hotel guests but it is a less desirable and possibly revenue-diminishing factor for hotels to deal with.

- **Can you predict if the customer is going to honor the reservation or cancel it?**

## **What is the target?**

- Booking status

## **What does one row represent?**

- Business

## **Is this a classification or regression problem?**

- Classification

## **How many features does the data have?**

- There are eighteen(18) features aside the target variable

## **How many rows are in the dataset?**

- 36275 rows

## **What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?**

In [ ]:
## Check for missing values
df_2.isnull().sum()

Booking_ID                              0
no_of_adults                            0
no_of_children                          0
no_of_weekend_nights                    0
no_of_week_nights                       0
type_of_meal_plan                       0
required_car_parking_space              0
room_type_reserved                      0
lead_time                               0
arrival_year                            0
arrival_month                           0
arrival_date                            0
market_segment_type                     0
repeated_guest                          0
no_of_previous_cancellations            0
no_of_previous_bookings_not_canceled    0
avg_price_per_room                      0
no_of_special_requests                  0
booking_status                          0
dtype: int64

**No missing values**

In [ ]:
## Have a look at the descriptive statistics
df_2.describe()

no_of_adults  no_of_children  no_of_weekend_nights  no_of_week_nights  \
count  36275.000000    36275.000000          36275.000000       36275.000000   
mean       1.844962        0.105279              0.810724           2.204300   
std        0.518715        0.402648              0.870644           1.410905   
min        0.000000        0.000000              0.000000           0.000000   
25%        2.000000        0.000000              0.000000           1.000000   
50%        2.000000        0.000000              1.000000           2.000000   
75%        2.000000        0.000000              2.000000           3.000000   
max        4.000000       10.000000              7.000000          17.000000   

       required_car_parking_space     lead_time  arrival_year  arrival_month  \
count                36275.000000  36275.000000  36275.000000   36275.000000   
mean                     0.030986     85.232557   2017.820427       7.423653   
std                      0.173281     85.930817      0.383836       3.069894   
min                      0.000000      0.000000   2017.000000       1.000000   
25%                      0.000000     17.000000   2018.000000       5.000000   
50%                      0.000000     57.000000   2018.000000       8.000000   
75%                      0.000000    126.000000   2018.000000      10.000000   
max                      1.000000    443.000000   2018.000000      12.000000   

       arrival_date  repeated_guest  no_of_previous_cancellations  \
count  36275.000000    36275.000000                  36275.000000   
mean      15.596995        0.025637                      0.023349   
std        8.740447        0.158053                      0.368331   
min        1.000000        0.000000                      0.000000   
25%        8.000000        0.000000                      0.000000   
50%       16.000000        0.000000                      0.000000   
75%       23.000000        0.000000                      0.000000   
max       31.000000        1.000000                     13.000000   

       no_of_previous_bookings_not_canceled  avg_price_per_room  \
count                          36275.000000        36275.000000   
mean                               0.153411          103.423539   
std                                1.754171           35.089424   
min                                0.000000            0.000000   
25%                                0.000000           80.300000   
50%                                0.000000           99.450000   
75%                                0.000000          120.000000   
max                               58.000000          540.000000   

       no_of_special_requests  
count            36275.000000  
mean                 0.619655  
std                  0.786236  
min                  0.000000  
25%                  0.000000  
50%                  0.000000  
75%                  1.000000  
max                  5.000000

In [ ]:
df_2.select_dtypes(exclude = np.number).columns

Index(['Booking_ID', 'type_of_meal_plan', 'room_type_reserved',
       'market_segment_type', 'booking_status'],
      dtype='object')

In [ ]:
df_2.describe(exclude = np.number)

Booking_ID type_of_meal_plan room_type_reserved market_segment_type  \
count       36275             36275              36275               36275   
unique      36275                 4                  7                   5   
top      INN24130       Meal Plan 1        Room_Type 1              Online   
freq            1             27835              28130               23214   

       booking_status  
count           36275  
unique              2  
top      Not_Canceled  
freq            24390

## **Summary 2**

- From the little analysis, there will be alot of modifications and challenges during the course of the work. 
- No much cleaning but alot of exploration is needed
- considering it is large dataset with many columns, some columns might make the model to underfit or overfit.